requirements.txt

```diff
@@ -37,6 +37,7 @@ psycopg2==2.8.5; sys_platform == 'win32' or python_version >= '3.8'
 pydot==1.4.1
 python-ldap==3.1.0; sys_platform != 'win32'
 PyPDF2==1.26.0
+pikepdf==3.0.0
```

Comment:
Please downgrade the version to 1.10.3
We must follow the [Debian](https://packages.debian.org/bullseye/python3-pikepdf)/[Ubuntu](https://packages.ubuntu.com/focal/python3-pikepdf) stable releases.
And as you see, the lowest is the Ubuntu package.
Also, if PyPDF2 is fully replaced, you can remove the PyPDF2 reference in the requirements.
The [Debian control file](https://github.com/odoo/odoo/blob/master/debian/control#L43) must me adapted as well as various other files:

- The packaging Debian Docker file https://github.com/odoo/odoo/blob/8b7d84064634ccef97e93a4a7f13344cba925636/setup/package.dfdebian#L43
- The rpm packaging Docker file https://github.com/odoo/odoo/blob/8b7d84064634ccef97e93a4a7f13344cba925636/setup/package.dffedora#L16
- The rpm bdist requirements: https://github.com/odoo/odoo/blob/8b7d84064634ccef97e93a4a7f13344cba925636/setup.cfg#L33
- And finally the src setup https://github.com/odoo/odoo/blob/8b7d84064634ccef97e93a4a7f13344cba925636/setup.py#L44

Reply:
@d-fence I tried to downgrade the pikepdf to 1.10.3.
But I need to add attachments to pdf using pikepdf.
In 1.10.3, I have to an internal function [`_attach`](https://github.com/pikepdf/pikepdf/issues/209). The function is not recommended by the author and removed from 3.0.0
In 3.0.0, I can use a class [`AttachedFileSpec`](https://pikepdf.readthedocs.io/en/latest/topics/attachments.html) which is introduced from 3.0.0
Since adding attachments is offically supported from 3.0.0, I think it would be better to use the newer version?

Reply:
I'm sorry to say no. It will not work for those who use the plain deb package or those who rely on python package of their distribution.

Reply:
> In 3.0.0, I can use a class [`AttachedFileSpec`](https://pikepdf.readthedocs.io/en/latest/topics/attachments.html) which is introduced from 3.0.0

You should make your code support both version, checking if AttachedFileSpec exists and fallbacking on _attach. It's crappy but needed in this case I think.

Reply:
> You should make your code support both version, checking if AttachedFileSpec exists and fallbacking on _attach. It's crappy but needed in this case I think.

Thanks for the proposal. I will try and see if my codes works for all versions